dataland-frontend/tests/e2e/specs/data-ownership/DataOwnershipUserJourney.ts

```diff
@@ -0,0 +1,85 @@
+import { describeIf } from "@e2e/support/TestUtility";
+import { admin_name, admin_pw, reader_name, reader_pw } from "@e2e/utils/Cypress";
+import { getKeycloakToken, login, logout } from "@e2e/utils/Auth";
+import {
+  CompanyDataControllerApi,
+  type CompanyDataOwners,
+  Configuration,
+  DataTypeEnum,
+  type PathwaysToParisData,
+} from "@clients/backend";
+import { generateDummyCompanyInformation, uploadCompanyViaApi } from "@e2e/utils/CompanyUpload";
+import { uploadFrameworkData } from "@e2e/utils/FrameworkUpload";
+import { type FixtureData, getPreparedFixture } from "@sharedUtils/Fixtures";
+import { ARRAY_OF_FRAMEWORKS_WITH_UPLOAD_FORM } from "@/utils/Constants";
+
+let p2pFixtureForTest: FixtureData<PathwaysToParisData>;
+before(function () {
+  cy.fixture("CompanyInformationWithP2pPreparedFixtures").then(function (jsonContent) {
+    const preparedFixturesP2p = jsonContent as Array<FixtureData<PathwaysToParisData>>;
+    p2pFixtureForTest = getPreparedFixture("P2p-dataset-with-no-null-fields", preparedFixturesP2p);
+  });
+});
+
+describeIf(
+  "As a user, I expect to be able to upload data for one company for which iam data owner",
+  {
+    executionEnvironments: ["developmentLocal", "ci", "developmentCd"],
+  },
+  function (): void {
+    beforeEach(() => {
+      cy.ensureLoggedIn(admin_name, admin_pw);
+    });
+
+    it("Upload a company, set a user as the data owner and then verify that the upload pages are displayed for that user", () => {
+      const uniqueCompanyMarker = Date.now().toString();
+      const testCompanyName = "Company-Created-In-Data-Owner-Test-" + uniqueCompanyMarker;
+      getKeycloakToken(admin_name, admin_pw).then((token: string) => {
+        return uploadCompanyViaApi(token, generateDummyCompanyInformation(testCompanyName)).then((storedCompany) => {
+          return uploadFrameworkData(
+            DataTypeEnum.P2p,
+            token,
+            storedCompany.companyId,
+            "2021",
+            p2pFixtureForTest.t,
+          ).then(() => {
+            void postDataOwner(token, "18b67ecc-1176-4506-8414-1e81661017ca", storedCompany.companyId);
```

Comment:
Wouldn't it make more sense to wait until the postDataOwner-call is finished before continuing in the test?

Reply:
I've added an intercept/wait block. Is that what you had in mind?

Reply:
Ah ok verstehe.

Ja das geht theoretisch auch.  Jedoch irritiert es mich dass der intercept "getDataOwner" heißt, wenn du doch einen postest (oder?).


Alternativ könntest du hier auch einfach promises chainen.   Also du sagst ihm quasi dass er erst weitermachen soll, wenn der data-owner erfolgreich gepostet wurde:

![image](https://github.com/d-fine/Dataland/assets/99255296/bf50c1e0-f2d6-417e-992a-c3c63a4289f9)



Also für mich wäre beides ok. Entweder:

a) Du benennst den intercept um

oder


b) du chainst es wie vorgeschlagen

Reply:
Hab jetzt einfach den intercept namen geändert